react/common/utils.tsx

```diff
@@ -381,6 +386,28 @@ export function prepareHistoryData(comment: any, request: any) {
       ),
       active: 1,
     },
+    {
+      status: statusHistoryTimeline.processing,
+      text: (
+        <span>
+          <FormattedMessageFixed id={messages.processing.id} />
+        </span>
+      ),
+      step: 1,
+      comments: comment.filter(
+        (item: any) => item.status === requestsStatuses.processing
+      ),
+      active:
+        request.status === requestsStatuses.processing ||
```

Comment:
I think there is some room here to make a reusable function too. For every single step, we are comparing the previous status and I can see it leading to some bugs in the future (e.g. a new status is added and we don't include it in the comparison). Having the logic in a helper function, might be easier to update it. What do you think?

Reply:
yeah that's true! i will make a helper function for it